src/decorators/BIPComponentTypeDecorator/DiagramDesigner/BIPComponentTypeDecorator.DiagramDesignerWidget.js

```diff
@@ -78,7 +87,44 @@ define([
     BIPComponentTypeDecorator.prototype.$DOMBase = $(BIPComponentTypeDecoratorTemplate);
 
     BIPComponentTypeDecorator.prototype.on_addTo = function () {
-        var self = this;
+        var self = this,
+            gmeID = self._metaInfo[CONSTANTS.GME_ID],
+            client = this._control._client;
+
+        function eventHandler(events) {
+            var nodeObj;
+
+            self.membersInfo = [];
+            self.implementersInfo = [];
+
+            nodeObj = client.getNode(gmeID);
+
+            if (!nodeObj || (nodeObj.getValidSetNames().indexOf(nodePropertyNames.Sets.associatedWith) === -1 && nodeObj.getValidSetNames().indexOf(nodePropertyNames.Sets.implementedBy) === -1)) {
+                return;
+            }
+            if (nodeObj.getValidSetNames().indexOf(nodePropertyNames.Sets.associatedWith) > -1) {
+                nodeObj.getMemberIds(nodePropertyNames.Sets.associatedWith)
+                    .forEach(function (id) {
+                        var memberNode = client.getNode(id);
```

Comment:
I would add a guard around the memberNode.
There is a potential race condition between `eventHandler` and the `update`. In the case where a new member was added and the `eventHandler` is called before the the `update`. The memberIds might include paths that aren't in a territory and not available. Without the guard there could be an exception, with it the data will be corrected once `update` is called.
You can put the guard at `name: memberNode ? memberNode.getAttribute('name') : 'N/A'`